### PR TITLE
feature/ELEC-482: replace config with envsafe (backend)

### PR DESCRIPTION
**Pull request by Merlte**
- removed config
- added env.ts in /src
- updated the dockerfile as well

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -44,6 +44,6 @@
     "prettier": "2.8.7",
     "ts-jest": "29.0.5",
     "tsx": "^3.12.6",
-    "typescript": "^5.0.2"
+    "typescript": "^4.3.0"
```
Comment (aimenhamed) [inline backend/package.json:47]: was there a reason for this change?

File: backend/src/interfaces/IConfig.ts [hunk 2]
```diff
@@ -5,12 +5,3 @@ export interface IApiConfig {
 export interface IConfig {
```
Comment (aimenhamed) [inline backend/src/interfaces/IConfig.ts:5]: we can probably delete this entire file right?

**Review by aimenhamed — CHANGES_REQUESTED** [review]

File: backend/src/interfaces/IConfig.ts (hunk 2 above)
Comment (Merlte) [reply backend/src/interfaces/IConfig.ts:5]: good point, I'll delete it right away!

File: backend/package.json (hunk 1 above)
Comment (aimenhamed) [reply backend/package.json:47]: yeah i think latest versioning is the practice we follow coz of the github bot settings.

**Review by lhjt — COMMENTED** [review]
Do these changes introduce or change any of the existing environment variables that are currently in use?

**Comment by aimenhamed** [comment]
> Do these changes introduce or change any of the existing environment variables that are currently in use?

this particular pr isn't changing any.

**Review by lhjt — APPROVED** [review]

File: backend/package.json (hunk 1 above)
Comment (Merlte) [reply backend/package.json]: it's been done btw

**Review by aimenhamed — APPROVED** [review]

**Pull request merged**